test/unit/Identifier/IdentifierTest.php

```diff
@@ -44,4 +47,73 @@ public function testIsTypesForFunction() : void
         self::assertFalse($identifier->isClass());
         self::assertTrue($identifier->isFunction());
     }
+
+    public function testGetNameForClosure() : void
+    {
+        $identifier = new Identifier(ReflectionFunctionAbstract::CLOSURE_NAME, new IdentifierType(IdentifierType::IDENTIFIER_FUNCTION));
+        self::assertSame(ReflectionFunctionAbstract::CLOSURE_NAME, $identifier->getName());
+    }
+
+    public function testGetNameForAnonymousClass() : void
+    {
+        $identifier = new Identifier(ReflectionClass::ANONYMOUS_CLASS_NAME_PREFIX . ' filename.php', new IdentifierType(IdentifierType::IDENTIFIER_CLASS));
+        self::assertStringStartsWith(ReflectionClass::ANONYMOUS_CLASS_NAME_PREFIX, $identifier->getName());
+    }
+
+    public function testGetNameForWildcard() : void
+    {
+        $identifier = new Identifier(Identifier::WILDCARD, new IdentifierType(IdentifierType::IDENTIFIER_CLASS));
+        self::assertSame(Identifier::WILDCARD, $identifier->getName());
+    }
+
+    public function validNamesProvider() : array
+    {
+        return [
+            ['Foo', 'Foo'],
+            ['\Foo', 'Foo'],
```

Comment:
Should we even accept `\\` as a leading character?

I think it would be nice to start educating users about this, but it's a possible BC break.

Thoughts?

Reply:
The leading `\\` is accepted, there's `ltrim` already.

Reply:
Yes, I was wondering whether we should tell users not to use it.

In strings, we are always talking about FQCN, so `\\` is redundant, and it can cause harm if not filtered.

We do it here, but the user should likely not keep passing it around that way in their own domain either.

Reply:
Or do you think to accept only names with leading `\\`? It would not be compatible with internal reflection and other libraries.

Reply:
> It would not be compatible with internal reflection and other libraries.

That is a fair point, and I think it clears the discussion point.